polytracker/src/passes/polytracker_pass.cpp

```diff
@@ -589,6 +600,8 @@ bool PolytrackerPass::runOnModule(llvm::Module &mod) {
         for (auto &inst : bb) {
           if (auto *BI = llvm::dyn_cast<llvm::BranchInst>(&inst)) {
             visitBranchInst(*BI);
+          } else if (auto *SI = llvm::dyn_cast<llvm::SwitchInst>(&inst)) {
```

Comment:
We can avoid the whole `for(auto &inst : bb){...}` by using `visit(bb->getTerminator())`.

Reply:
Great idea! We can't do exactly as you write, but we don't need to loop. If we use the visitor we would also get e.g. the ret-instr visitation and that is not desired.

Reply:
True, my bad! Does it matter when we visit branches and switches?

Reply:
We would add instrumentation for the ret instruction and that typically instruments the instruction with a call to a method to log function exits.  I'll make a slight variation.